MapleServer2/PacketHandlers/Game/InteractObjectHandler.cs

```diff
@@ -56,6 +56,14 @@ private static void HandleStart(GameSession session, PacketReader packet)
         private static void HandleUse(GameSession session, PacketReader packet)
         {
             string uuid = packet.ReadMapleString();
+
+            if(uuid.StartsWith("BillBoard")) // Possible temp solution?
+            {
+                IFieldObject<AdBalloon> balloon = session.FieldManager.State.Balloons[uuid];
+                session.Send(PlayerHostPacket.AdBalloonWindow(balloon));
+                return;
+            }
+
```

Comment:
Highlighting this portion for conversation:
At the moment, we're checking if an interact object exists in the field by MapEntityStorage. Of course, this wont work for balloons because they're user-placed objects. 
My currently solution right now to differentiate the two is by the name. _IDEALLY_ I think we should go by InteractObjectType.
Instead of searching MapEntityStorage for the InteractObject, should we instead search the Field?

Reply:
The best would be to rework _**InteractObjectHandler**_ to use `FieldState` instead of **_MapEntityStorage_**. Then with the enum _InteractObjectType_ it should also handle future objects that are added through the instance like instance portals (not sure the name, portals that help you navigate quests)

Reply:
The best would be to rework InteractObjectHandler to use FieldState instead of MapEntityStorage. Then with the enum InteractObjectType it should also handle future objects that are added through the instance like instance portals (not sure the name, portals that help you navigate quests)

Reply:
Yeah that's what I was thinking. And have a switch case for the InteractObjectType.
Would this mean AdBalloons need to be combined with "normal" InteractObject?

Reply:
I think it's the best way unless there is a reason to AdBallons be its own `type`. If InteractObject and AdBalloon have way too different implementations then they should be separate, which I don't think it's the case